### ISSUE TITLE
Error when running finetuning

### ISSUE THREAD

**Issue by matt9704** [google-research/pegasus#43]
Hi! I follow your readme instruction and run the code on google colab, but errors show up when running the finetuning part:
`!python3 pegasus/bin/train.py --params=aeslc_transformer \
--param_overrides=vocab_filename=ckpt/pegasus_ckpt/c4.unigram.newline.10pct.96000.model \
--train_init_checkpoint=ckpt/pegasus_ckpt/model.ckpt-1500000 \
--model_dir=ckpt/pegasus_ckpt/aeslc`

This error report shows up again and again in the output field:
``2020-07-02` 18:53:50.204708: I tensorflow/core/common_runtime/bfc_allocator.cc:929] Stats: 
Limit:                 11330115994
InUse:                 10814302464
MaxInUse:              10881414400
NumAllocs:                    4881
MaxAllocSize:            565956352

2020-07-02 18:53:50.204943: W tensorflow/core/common_runtime/bfc_allocator.cc:424] *************************************************************************************************___
2020-07-02 18:53:50.208784: W tensorflow/core/framework/op_kernel.cc:1651] OP_REQUIRES failed at softmax_op_gpu.cu.cc:157 : Resource exhausted: OOM when allocating tensor with shape[8,16,1024,1024] and type float on /job:localhost/replica:0/task:0/device:GPU:0 by allocator GPU_0_bfc
ERROR:tensorflow:Error recorded from training_loop: 2 root error(s) found.
  (0) Resource exhausted: OOM when allocating tensor with shape[8,16,1024,1024] and type float on /job:localhost/replica:0/task:0/device:GPU:0 by allocator GPU_0_bfc
	 [[node encoder/layer_5/self_attention/Softmax (defined at /usr/local/lib/python3.6/dist-packages/tensorflow_core/python/framework/ops.py:1748) ]]
Hint: If you want to see a list of allocated tensors when OOM happens, add report_tensor_allocations_upon_oom to RunOptions for current allocation info.

	 [[softmax_cross_entropy_loss/value/_9835]]
Hint: If you want to see a list of allocated tensors when OOM happens, add report_tensor_allocations_upon_oom to RunOptions for current allocation info.

  (1) Resource exhausted: OOM when allocating tensor with shape[8,16,1024,1024] and type float on /job:localhost/replica:0/task:0/device:GPU:0 by allocator GPU_0_bfc
	 [[node encoder/layer_5/self_attention/Softmax (defined at /usr/local/lib/python3.6/dist-packages/tensorflow_core/python/framework/ops.py:1748) ]]
Hint: If you want to see a list of allocated tensors when OOM happens, add report_tensor_allocations_upon_oom to RunOptions for current allocation info.

0 successful operations.
0 derived errors ignored.

Original stack trace for 'encoder/layer_5/self_attention/Softmax':
  File "pegasus/bin/train.py", line 94, in <module>
    tf.app.run(main)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/platform/app.py", line 40, in run
    _run(main=main, argv=argv, flags_parser=_parse_flags_tolerate_undef)
  File "/usr/local/lib/python3.6/dist-packages/absl/app.py", line 299, in run
    _run_main(main, args)
  File "/usr/local/lib/python3.6/dist-packages/absl/app.py", line 250, in _run_main
    sys.exit(main(argv))
  File "pegasus/bin/train.py", line 89, in main
    max_steps=train_steps)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_estimator/python/estimator/tpu/tpu_estimator.py", line 3030, in train
    saving_listeners=saving_listeners)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_estimator/python/estimator/estimator.py", line 370, in train
    loss = self._train_model(input_fn, hooks, saving_listeners)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_estimator/python/estimator/estimator.py", line 1161, in _train_model
    return self._train_model_default(input_fn, hooks, saving_listeners)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_estimator/python/estimator/estimator.py", line 1191, in _train_model_default
    features, labels, ModeKeys.TRAIN, self.config)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_estimator/python/estimator/tpu/tpu_estimator.py", line 2857, in _call_model_fn
    config)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_estimator/python/estimator/estimator.py", line 1149, in _call_model_fn
    model_fn_results = self._model_fn(features=features, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_estimator/python/estimator/tpu/tpu_estimator.py", line 3126, in _model_fn
    features, labels, is_export_mode=is_export_mode)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_estimator/python/estimator/tpu/tpu_estimator.py", line 1663, in call_without_tpu
    return self._call_model_fn(features, labels, is_export_mode=is_export_mode)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_estimator/python/estimator/tpu/tpu_estimator.py", line 1994, in _call_model_fn
    estimator_spec = self._model_fn(features=features, **kwargs)
  File "/content/pegasus/pegasus/params/estimator_utils.py", line 105, in model_fn
    loss, outputs = model_params.model()(features, training)
  File "/content/pegasus/pegasus/models/transformer.py", line 91, in __call__
    context = self._encode(features, training)
  File "/content/pegasus/pegasus/models/transformer.py", line 72, in _encode
    None, None)
  File "/content/pegasus/pegasus/layers/transformer_block.py", line 100, in stack
    decode_i=decode_i)
  File "/content/pegasus/pegasus/layers/transformer_block.py", line 65, in __call__
    y_BxIxD, bias_BxIxI, training, cache=cache, decode_i=decode_i)
  File "/content/pegasus/pegasus/layers/attention.py", line 94, in __call__
    x, x, bias, training, cache=cache, decode_i=decode_i)
  File "/content/pegasus/pegasus/layers/attention.py", line 78, in __call__
    alignment_BxHxIxM = tf.nn.softmax(logits_BxHxIxM)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/util/deprecation.py", line 507, in new_func
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/ops/nn_ops.py", line 2958, in softmax
    return _softmax(logits, gen_nn_ops.softmax, axis, name)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/ops/nn_ops.py", line 2891, in _softmax
    return compute_op(logits, name=name)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/ops/gen_nn_ops.py", line 11376, in softmax
    "Softmax", logits=logits, name=name)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/framework/op_def_library.py", line 794, in _apply_op_helper
    op_def=op_def)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/util/deprecation.py", line 507, in new_func
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/framework/ops.py", line 3357, in create_op
    attrs, op_def, compute_device)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/framework/ops.py", line 3426, in _create_op_internal
    op_def=op_def)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/framework/ops.py", line 1748, in __init__
    self._traceback = tf_stack.extract_stack()
``

**Comment by darienacosta** [comment]
*Resource exhausted: OOM when allocating tensor with shape[8,16,1024,1024] *
I think your GPU doesn't have enough memory to compute the models. What GPU are you using?

**Comment by matt9704** [comment]
> *Resource exhausted: OOM when allocating tensor with shape[8,16,1024,1024] *
> I think your GPU doesn't have enough memory to compute the models. What GPU are you using?

I think it's K80.

**Comment by darienacosta** [comment]
You can also try batching your training with batch_size=1 like in the following
Duplicate of #39

**Comment by matt9704** [comment]
Thanks I'll have a try.
